### DISCUSSION TITLE
phalcon 5 - Checkbox is not Checked

### DISCUSSION THREAD

**Discussion by mikx33** [phalcon/cphalcon#15932] [General]
used like this
`{{ form.render('vatEnabled', ["class": "custom-control-input", "id": "vatEnabled", "checked" : "checked"]) }}`

result:
`<input type="checkbox" id="vatEnabled" name="vatEnabled" value="1" class="custom-control-input">`

**Comment by lieshanhe** [comment]
It's also occurred to me. besides that, there is an issue when the form tries to bind $_POST data with the model instance.

.volt
``
 {{ form.renderDecorated('password') }}
``

Controller
```
if ($this->request->isPost()) {
    $post = $this->request->getPost();
    $form->bind($post, $model);
   ........
} else {
   $form->setEntity($model);
}
```

Form:
``
$this->add(
            (new Password('password',[
                'placeholder' => 'Password',
            ]))->setLabel('Password')
        );
``


Model:
```
public function setPassword($password)
{
        if ($password) {
            $security = DI::getDefault()->get('security');
            $this->password = $security->hash($password);
        }
}

public function getPassword()
{
       return ''; // We don't need hash of password. Just return empty string.
}
```

The code above won't save any string to the password column. But if I updated the getPassword(); it works.
```
public function getPassword()
{
       return  $this->password;
}
```

**Comment by alimo2** [comment]
It's not working !

Any problem in:
https://github.com/phalcon/cphalcon/blob/master/phalcon/Html/Helper/Input/Checkbox.zep

**Comment by punneke** [comment]
Having the same issue. It's reported here: https://github.com/phalcon/cphalcon/issues/15959
I also posted a workaround that works for me.

```php
$status = new Check('status', [
    'value' => 1,
]);
if (!is_null($entity)) {
    $status->setAttribute('checked', $entity->status);
}
$status->setLabel('Status');
$this->add($status);
```

I think the issue lies in the fact that the value in `$entity->active` is `1` as an integer and not `'1'` a string. 
This probably changed in Phalcon 5 because all database integers used to be returned as a string and now they are integers.

So the check on line 127 in https://github.com/phalcon/cphalcon/blob/master/phalcon/Html/Helper/Input/Checkbox.zep#L127 fails because is strict with `===`.

**Comment by s-ohnishi** [comment]
@mikx33 If you define the following in a class that extends Forms:
``` PHP
$elm = new Element\Check('vatEnabled', ['value' => 1]);
$this->add($elm);
```
It should be specified as follows.
`{{ form.render('vatEnabled', ["class": "custom-control-input", "id": "vatEnabled", "checked" : 1]) }} `

Also, in .volt, if you call TagFactory directly, you need to specify it like this.
`{{ tag.inputCheckbox('testcb', 1, ['checked':'1']) }}`
or
`{{ tag.inputCheckbox('testcb', '1', ['checked':'1']) }}`

It is important to note that even if you pass a numeric value in the second argument of inputCheckbox (), it seems to be treated as a string internally, so the checked value must be given as a string.
